mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MailboxMapperTest.java

```diff
@@ -82,25 +82,60 @@ void setUp() {
     }
 
     @Test
-    void findMailboxByPathWhenAbsentShouldFail() throws MailboxException {
+    void findMailboxByPathWhenAbsentShouldFail() {
         assertThatThrownBy(() -> mailboxMapper.findMailboxByPath(MailboxPath.forUser(BENWA, "INBOX")))
             .isInstanceOf(MailboxNotFoundException.class);
     }
 
     @Test
-    void saveShouldPersistTheMailbox() throws MailboxException {
-        mailboxMapper.save(benwaInboxMailbox);
+    void createShouldPersistTheMailbox() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        mailboxMapper.create(benwaInboxMailbox);
+
+        MailboxAssert.assertThat(mailboxMapper.findMailboxByPath(benwaInboxPath)).isEqualTo(benwaInboxMailbox);
+        MailboxAssert.assertThat(mailboxMapper.findMailboxById(benwaInboxMailbox.getMailboxId())).isEqualTo(benwaInboxMailbox);
+    }
+
+    @Test
+    void createShouldThrowWhenMailboxAlreadyExists() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        mailboxMapper.create(benwaInboxMailbox);
+
+        Mailbox mailbox = new Mailbox(benwaInboxMailbox);
+        mailbox.setMailboxId(null);
+
+        assertThatThrownBy(() -> mailboxMapper.create(mailbox))
+            .isInstanceOf(MailboxExistsException.class);
+    }
+
+    @Test
+    void createShouldSetAMailboxIdForMailbox() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        MailboxId mailboxId = mailboxMapper.create(benwaInboxMailbox);
+
+        assertThat(mailboxId).isNotNull();
+    }
+
+    @Test
+    void createShouldThrowWhenMailboxIdNotNull() {
+        assertThatThrownBy(() -> mailboxMapper.create(benwaInboxMailbox))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void renameShouldPersistTheMailbox() throws MailboxException {
```

Comment:
idem, this sounds weird.

Reply:
The behavior of "rename" logic doesn't change, so it still creates and rename a mailbox. I will change the rename behavior, as I said, in the next PR #3108